solidity/contracts/token/libs/TokenRouter.sol

```diff
@@ -142,6 +142,9 @@ abstract contract TokenRouter is GasRouter, ITokenBridge {
         bytes32 _recipient,
         uint256 _amount
     ) public payable virtual returns (bytes32 messageId) {
+        // Prevent zero-amount transfers to avoid spam
+        require(_amount > 0, "TokenRouter: amount must be greater than 0");
```

Comment:
sending 0 might still be a valid transfer depending on what else is in the message. the anti-spam mechanic is the IGP fee which has to be paid to relay even if the message is 0 tokens